### PR TITLE
refactor: Use vars/RedHat_N.yml symlink for CentOS, Rocky, Alma wherever possible

### PR DESCRIPTION
**Pull request by richm**
We have a lot of requests to support Rocky and Alma in various system roles. The
first part of adding support is adding `vars/` files for these platforms. In
almost every case, for a given major version N, the vars file RedHat_N.yml can
be used for CentOS, Rocky, and Alma.  Rather than making a copy of the
RedHat_N.yml file, just use a symlink to reduce size and maintenance burden, and
standardize this across all system roles for consistency.

NOTE: OracleLinux is not a strict clone, so we are not going to do this for
OracleLinux at this time.  Support for OracleLinux will need to be done in
separate PRs. For more information, see
https://github.com/linux-system-roles/cockpit/issues/130

Note that there may be more work to be done to the role to fully support Rocky
and Alma.  Many roles have conditionals like this:

```yaml
some_var: "{{ 'some value' if ansible_distribution in ['CentOS', 'RedHat'] else 'other value' }}"
another_var: "{{ 'some value' if ansible_distribution in ['CentOS', 'Fedora', 'RedHat'] else 'other value' }}"

...

- name: Do something
  when: ansible_distribution in ['CentOS', 'RedHat']
  ...
- name: Do something else
  when: ansible_distribution in ['CentOS', 'Fedora', 'RedHat']
  ...
```

Adding Rocky and AlmaLinux to these conditionals will have to be done
separately. In order to simplify the task, some new variables are being
introduced:

```yaml
__$rolename_rh_distros:
  - AlmaLinux
  - CentOS
  - RedHat
  - Rocky

__$rolename_rh_distros_fedora: "{{ __$rolename_rh_distros + ['Fedora'] }}"

__$rolename_is_redhat_distro: "{{ ansible_distribution in __$rolename_rh_distros }}"
__$rolename_is_redhat_distro_fedora: "{{ ansible_distribution in __$rolename_rh_distros_fedora }}"
```

Then the conditionals can be rewritten as:

```yaml
some_var: "{{ 'some value' if __$rolename_is_redhat_distro else 'other value' }}"
another_var: "{{ 'some value' if __$rolename_is_redhat_distro_fedora else 'other value' }}"

...

- name: Do something
  when: __$rolename_is_redhat_distro | bool
  ...
- name: Do something else
  when: __$rolename_is_redhat_distro_fedora | bool
  ...
```

For tests - tests that use such conditionals will need to use `vars_files` or
`include_vars` to load the variables that are defined in
`tests/vars/redhat_clone_vars.yml`:

```yaml
vars_files:
  - vars/redhat_clone_vars.yml
```

We don't currently have CI testing for Rocky or Alma, so someone wanting to run
tests on those platforms would need to change the test code to use these.

### CODE REVIEW

**Comment by richm** [comment]
Also includes some improvements for updating existing PRs - update commit msg, PR title and body

**Comment by spetrosi** [comment]
I believe we remove symlinks when building RPM. We should ensure to keep symlinks that we add in this PR, and downstream I guess it doesn't matter to remove or keep much.

**Comment by richm** [comment]
> I believe we remove symlinks when building RPM. We should ensure to keep symlinks that we add in this PR, and downstream I guess it doesn't matter to remove or keep much.

ok - but I think it already works - some roles have been using symlinks like this for years - which is what gave me the idea to just do this for all roles

**Comment by richm** [comment]
@spetrosi this is ready for review - you can also check one of the role PRs like https://github.com/linux-system-roles/nbde_client/pull/181

File: playbooks/update_files.yml [hunk 1]
```diff
@@ -271,22 +359,30 @@
...
               else
-                git commit -s -F {{ update_files_commit_file | quote }}
-                force_flag="{{ __is_new | ternary('', '-f') }}"
-                if [ {{ lsr_dry_run | d('true') }} = false ]; then
-                  git push $force_flag -u origin {{ update_files_branch | quote }}
-                echo
-                  echo git push $force_flag -u origin {{ update_files_branch | quote }}
-                fi
+                # commit orig head, then amend, and ensure commit is signed
+                git commit -C ORIG_HEAD
```
Comment (spetrosi) [inline playbooks/update_files.yml:369]: Why is this needed?

**Review by spetrosi — APPROVED** [review]
other than 1 nitpick and one question lgtm

File: playbooks/update_files.yml (hunk 1 above)
Comment (richm) [reply playbooks/update_files.yml:369]: This is needed so that you can edit existing PRs

Comment (spetrosi) [reply playbooks/update_files.yml:369]: Right now I see, that's smart
